docs/source-2.0/guides/client-guidance/identity-auth/index.md

```diff
@@ -0,0 +1,416 @@
+
+This may be implemented like the following:
+
+```java
+public void resolveAuthScheme(OperationContext ctx) {
+    AuthSchemeResolver resolver = ctx.getAuthSchemeResolver();
+    MyServiceAuthSchemeResolverInput input = MyServiceAuthSchemeResolverInput.builder()
+        .operation(ctx.getOperationName())
+        .build();
+
+    List<AuthSchemeOption> options = resolver.resolveAuthScheme(input);
+    SelectedAuthScheme selected = selectAuthScheme(ctx, options);
+    if (selected == null) {
+        throw new OperationException("no available auth schemes");
+    }
+
+    ctx.setSelectedAuthScheme(selected);
+}
+
+private void selectAuthScheme(OperationContext ctx, List<AuthSchemeOption> options) {
+    MyServiceClientConfig config = ctx.clientConfig();
+
+    for (AuthScheme option : options) {
+        // condition 1: the client has an auth scheme with an id matching the
+        // one of this option
+        AuthScheme<?, ?> found = ctx.getAuthSchemes().stream()
+            .filter(it -> it.schemeId() == option.schemeId())
+            .findFirst();
+        if (found.isEmpty()) {
+            continue;
+        }
+
+        // condition 2: the client has an identity resolver configured for this auth scheme
+        //
+        // note that we are only checking whether there IS an identity resolver, not whether
+        // it can actually provide an identity
+        IdentityResolver resolver = scheme.getIdentityResolver(config);
+        if (resolver == null) {
+            continue;
+        }
+
+        return new SelectedAuthScheme(
+            found.get(),
+            option.identityProperties(),
+            option.signerProperties()
+        );
+    }
+
+    return null;
+}
+
+record SelectedAuthScheme(
+    AuthScheme<?, ?> scheme,
+    Object identityProperties,
+    Object signerProperties
+) {}
+```
+
+## Order of Operations
+
+The Smithy client should conduct the auth flow in an operation call as follows:
+
+1. Auth scheme resolution (AuthSchemeResolver::resolveAuthScheme) is called
+   **within** the retry loop. The selected auth scheme and identity / signer
+   properties are stored in operation context for future use.
+1. (Endpoint resolution)[TODO: NEED ANCHOR] is called. The signer properties
+   from the resolved endpoint are **merged** into the ones sourced from auth
+   scheme resolution.
+1. Retrieve the IdentityResolver from the previously-resolved AuthScheme
+   (AuthScheme::identityResolver). Identity resolution
+   (IdentityResolver::resolve) is called with the identity properties sourced from
+   scheme resolution.
+1. Retrieve the Signer from the previously-resolved AuthScheme
+   (AuthScheme::signer). Request signing (Signer::sign) is called with the
+   merged signer properties from scheme and endpoint resolution.
+
+## FAQ
+
+### Wow, this seems like a lot. Do I really need all of these abstractions just to decide how to set an Authorization header?
```

Comment:
Probably controversial but this felt absolutely worth stating IMO. I think a lot of client designers _will_ sort of stick to baking in a single transport type (probably HTTP). The language of the question is probably too colloquial ofc.